service/internal/fanoutconsumer/logs.go

```diff
@@ -80,3 +83,50 @@ func (lsc *logsConsumer) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
 	}
 	return errs
 }
+
+var _ connector.LogsRouter = (*logsRouter)(nil)
+
+type logsRouter struct {
+	consumer.Logs
+	consumers map[component.ID]consumer.Logs
+}
+
+func NewLogsRouter(cm map[component.ID]consumer.Logs) consumer.Logs {
```

Comment:
Confused why you need a new class here vs changing NewLogs to return this?

Reply:
This means that receiver constructors can assert `connector.LogsRouter` as well. Maybe this is ok but I wasn't expecting to enable this.

Reply:
It also turns out to be quite an overhaul of `NewLogs` and its tests, because we need to pass in `map[component.ID]consumer.Logs`, instead of just `[]consumer.Logs`. Doable, but seems to generate enough unnecessary code that it's worse than having a second class in my opinion.

Reply:
Are you ok with leaving these as separate structs?